indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx

```diff
@@ -140,7 +143,8 @@ class BookingEdit extends React.Component {
         const availabilityData = availability[0][1];
         const conflicts = _.fromPairs(newDateRange.map((day) => {
             const allConflicts = availabilityData.conflicts[day] || [];
-            return [day, allConflicts.filter((c) => c.reservation.id !== bookingId)];
+            const validDay = !(day in calendar.data.cancellations || day in calendar.data.rejections);
```

Comment:
in the `filter()` below you always reject all entries if this is false, so you can simply return  early without even iterating over `allConflicts`:

```js
if (day in calendar.data.cancellations || day in calendar.data.rejections) {
    return [day, []];
}
```